iteration2: refactor code base

binarytrails opened the pull request.

joshpham97 committed:
Remove play button/Start timer only when a game is started/Refactor game difficulty selection

joshpham97 committed:
UI adjustments and refactor

Added a UI for when the application first run (including moving frame initialization outside of board_ui())
Display the timer only when a game is in progress
Adjust borders of UI elements

joshpham97 committed:
Refactor the code and fixed merge conglicts and commented out TestGameController

joshpham97 committed:
Fixed test bug and renamed App Controller to Game Controller for consistency

joshpham97 committed:
Fixed more test cases

changlmasaki committed:
uncommented all, refactored and fixed errors

changlmasaki reviewed (commented):
fixed the game controller

changlmasaki committed:
Update TestGameController.java

joshpham97 committed:
Apply the correct flow and changed the variable name

joshpham97 committed:
Refactor

Merge BoardController with GameController
Rename Button Menu to Menu Bar
Rename BoardCell to Cell
